server/src/main/java/org/elasticsearch/search/aggregations/metrics/cardinality/CardinalityAggregator.java

```diff
@@ -181,6 +200,56 @@ public void close() {
         }
     }
 
+    // FIXME: This "RollupCollector" was written as a hack to prove that deserializing the HLLs might work
+    private static class RollupCollector extends Collector {
+
+        private final HyperLogLogPlusPlus counts;
+        private final SortedBinaryDocValues rollups;
+
+        RollupCollector(HyperLogLogPlusPlus counts, SortedBinaryDocValues rollups) {
+            this.counts = counts;
+            this.rollups = rollups;
+        }
+
+        @Override
+        public void collect(int doc, long bucketOrd) throws IOException {
+            // Each binary blob is in the SortedBinaryDocValues object, so we just advance along deserialize, and merge.
+            if (rollups.advanceExact(doc)) {
+                BytesRef bytes = rollups.nextValue();
+                // assert bytes.length > 0 : "Decoded HLL had no bytes";
+                // Originally, we created the copy this way, but when I read more about
+                // what the BytesRef actually is, I realized it's totally wrong.
+                // byte[] hllBytes = bytes.bytes;
+                // ByteArrayInputStream bais = new ByteArrayInputStream(hllBytes);
+                // The right way to do it: use the alternative constructor of ByteArrayInputStream:
+
+                //byte[] hllBytes = BytesRef.deepCopyOf(bytes).bytes;
+                byte[] hllBytesCopy = BytesRef.deepCopyOf(bytes).bytes;
```

Comment:
In a couple of commits, I thought the right way to do this was to pass `bytes.bytes, bytes.offset, bytes.length` to the `ByteArrayInputStream` constructor, since via Lucene talks I discovered that the buffer underneath the `BytesRef` is not just something you can pass around freely. But, when I tried to make use of a copy or do that trick, I hit other issues, so I tabled it for now.